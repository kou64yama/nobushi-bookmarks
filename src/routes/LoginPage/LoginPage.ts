import Vue from 'vue';
import Component from 'vue-class-component';
import { mapState } from 'vuex';

@Component({
  computed: {
    ...mapState('auth', ['loading']),
  },
})
export default class LoginPage extends Vue {
  public email = '';
  public password = '';

  public mounted() {
    this.$store.dispatch('document/setTitle', 'Log in');
  }

  public signIn() {
    this.$store.dispatch('auth/signInWithEmailAndPassword', {
      email: this.email,
      password: this.password,
    });
  }
}
