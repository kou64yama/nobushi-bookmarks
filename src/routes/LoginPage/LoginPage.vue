<template>
  <v-container fill-height justify-center>
    <v-progress-circular indeterminate size="64" color="primary" v-if="loading"/>
    <v-layout justify-center align-center v-else>
      <v-flex xs12 sm10 md8 lg6 xl4>
        <h1>Log in</h1>
        <v-text-field
          label="Email"
          v-model="email"
          @keyup.enter="signIn"
        />
        <v-text-field
          label="Password"
          type="password"
          v-model="password"
          @keyup.enter="signIn"
        />
        <v-btn block color="primary" @click.native="signIn">
          Log in
        </v-btn>
      </v-flex>
    </v-layout>
  </v-container>
</template>

<script lang="ts" src="./LoginPage.ts"></script>
