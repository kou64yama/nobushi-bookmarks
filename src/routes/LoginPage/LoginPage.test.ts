import Vuex, { Store, Dispatch } from 'vuex';
import { mount, createLocalVue } from '@vue/test-utils';
import LoginPage from './LoginPage.vue';

const localVue = createLocalVue();

localVue.use(Vuex);

describe('LoginPage', () => {
  let store: Store<any>;
  let dispatch: jest.Mock<Dispatch>;

  beforeEach(() => {
    store = new Vuex.Store({});
    dispatch = store.dispatch = jest.fn<Dispatch>();
  });

  it('should render correctly when loading', () => {
    const wrapper = mount(LoginPage, {
      localVue,
      store,
      stubs: {
        VContainer: true,
        VProgressCircular: true,
        VLayout: true,
        VFlex: true,
        VTextField: true,
        VBtn: true,
      },
      computed: {
        loading: () => true,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(dispatch.mock.calls[0][0]).toEqual('document/setTitle');
  });

  it('should render correctly when not loading', () => {
    const wrapper = mount(LoginPage, {
      localVue,
      store,
      stubs: {
        VContainer: true,
        VProgressCircular: true,
        VLayout: true,
        VFlex: true,
        VTextField: true,
        VBtn: true,
      },
      computed: {
        loading: () => false,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
    expect(dispatch.mock.calls[0][0]).toEqual('document/setTitle');
  });

  it('should dispatch auth/signInWithEmailAndPassword when button is clicked', () => {
    const wrapper = mount(LoginPage, {
      localVue,
      store,
      stubs: {
        VContainer: true,
        VProgressCircular: true,
        VLayout: true,
        VFlex: true,
        VTextField: true,
        VBtn: true,
      },
      computed: {
        loading: () => false,
      },
    });

    wrapper.vm.$data.email = 'user@example.com';
    wrapper.vm.$data.password = 'secret';
    wrapper.find('vbtn-stub').trigger('click');
    expect(dispatch).toHaveBeenLastCalledWith(
      'auth/signInWithEmailAndPassword',
      {
        email: 'user@example.com',
        password: 'secret',
      },
    );
  });
});
