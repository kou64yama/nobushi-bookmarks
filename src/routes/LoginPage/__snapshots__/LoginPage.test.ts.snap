// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage should render correctly when loading 1`] = `
<vcontainer-stub
  fill-height=""
  justify-center=""
>
  <vprogresscircular-stub
    color="primary"
    indeterminate=""
    size="64"
  />
</vcontainer-stub>
`;

exports[`LoginPage should render correctly when not loading 1`] = `
<vcontainer-stub
  fill-height=""
  justify-center=""
>
  <vlayout-stub
    align-center=""
    justify-center=""
  >
    <vflex-stub
      lg6=""
      md8=""
      sm10=""
      xl4=""
      xs12=""
    >
      <h1>
        Log in
      </h1>
       
      <vtextfield-stub
        label="Email"
      />
       
      <vtextfield-stub
        label="Password"
        type="password"
      />
       
      <vbtn-stub
        block=""
        color="primary"
      >
        
        Log in
      
      </vbtn-stub>
    </vflex-stub>
  </vlayout-stub>
</vcontainer-stub>
`;
